### ISSUE TITLE
IRC, Slack or Gitter

### ISSUE THREAD

**Issue by squeaky-pl** [squeaky-pl/japronto#7]
I see a lot of you have questions and valuable feedback. I am planning on opening some chat room to address those faster. I have no personal preference when it comes to IRC, Slack, Gitter or something else. I would like to hear which one you want to use.

**Comment by JeffSpies** [comment]
+1 for Gitter. Free without limits to history (as opposed to Slack) and seems to be more accessible than IRC these days.

**Comment by kianxineki** [comment]
+1 for Gitter
@JeffSpies I think there was a gateway to use gitter from an irc client

**Comment by r0fls** [comment]
+1 gitter, it's quite awesome

**Comment by luizberti** [comment]
A friend of mine runs a [popular Open Source project](https://github.com/developit/preact), and has experimented with all of them (Gitter, Discord, etc). From his experience, Gitter had always been a stagnate, silent place (as has been my experience with every Gitter I have ever joined). There was essentially no community participation and people would just hop in to ask questions, then leave.

A month ago he switched to Slack and saw an immediate, and very significant boost in community engagement. Said it was a great decision and never looked back again.

Just my 2¢ on this.

**Comment by santagada** [comment]
I hate having to connect to a bunch of slack servers, so for me gitter or irc are easier. JavaScript browser projects are outliers in this respect I think.

**Comment by ibakirov** [comment]
+1 Gitter

**Comment by wgwz** [comment]
1. irc 2. gitter 3. slack bc i'm opinionated that way

**Comment by kianxineki** [comment]
@wgwz 
slack: https://get.slack.help/hc/en-us/articles/201727913-Connect-to-Slack-over-IRC-and-XMPP
gitter: https://github.com/WICG/admin/wiki/Connecting-to-Gitter-over-IRC

**Comment by wgwz** [comment]
@kianxineki it's more of a protocol opinion ;)

**Comment by jacobbridges** [comment]
I agree with @luizberti that gitter chats seem to always be dead while Slack chats are more of a community, but gitter is free and easily accessible while Slack has limited history and requires more steps for registration. So I'm for gitter.

**Comment by squeaky-pl** [comment]
So after all the arguments brought here I am leaning towards IRC, I looked into other projects' Gitter channels for a while but it felt dead indeed and non-interactive at all. I kind of like Slack but its business model gets in a way when it comes to open source.

**Comment by wgwz** [comment]
i found the use of this logging tool, https://botbot.me/, in the pocoo channel to be really helpful. especially early on when i didn't have any logging set up myself. if i want to read up on what's been going on, i'll just check in there.

**Comment by luizberti** [comment]
The message cap does look like an issue at first, but quite honestly I've been using Slack for a long time across many teams that use the free tier and I never actually stumbled upon it. The only thing that might be problematic concerning it is if we were using it to store permanent decisions about the project.

IMHO, as long as we keep all formal discussion and official stuff here in the issue tracker, and use Slack as a means to engage as a community and discuss other less important aspects of development -- in short, keep discussions ephemeral and move important things to the tracker _(as it should be done anyways, no matter what chat service gets picked)_ -- Slack's message cap will never be an issue.

I can't remember a single time in any Gitter chatroom ever where searching back further than 10k messages was something I needed to do. Infinite scroll back is a mostly useless feature that people seem to pay too much attention to.

In any case, this is just my opinion: YMMV

**Comment by squeaky-pl** [comment]
I decided to try Slack, but is there a way I can disable the invitation madness? I tried adding domain names into whitelisted e-mail domains but it won't allow me add generic domains like gmail.com.

**Comment by squeaky-pl** [comment]
I think I gave up on Slack. It has all the bells and whistles, I used it a lot in the past companies and it's great but it's kind of geared towards small business and closed communities.

I don't want anybody who wants to ask a simple question go through the invitation process. Also it's a no go for many concerned people in the open source community.

We can do fine with IRC. It's simple, lightweight and anyone who was previously involved with open source knows how to use it. I myself am a heavy IRC user (And sorry for the bias here but I'm not particularly excited about running fully blown JavaScript client to just exchange text messages which is what you do 95% of time on services like Slack) . I thought it makes sense to look at the alternatives (that's why I started the whole discussion) but after having a closer look it feels more painful for not so much gain.

I set up a channel as in the guide that @wgwz shared. It's on Freenode and the channel name is #japronto.

**Comment by squeaky-pl** [comment]
I also requested botbot.me logging but it's gonna take some time:

```
Your Request Has Been Received

Adding channels is a manual process and we prioritize logging large active channels. We do our best to get to new requests in a timely fashion, but we typically have a large backlog. It may be weeks before anyone follows up with you. Please be patient!
```

**Comment by squeaky-pl** [comment]
I added Gitter integration with IRC. Anyone who prefers Gitter can join here: https://gitter.im/japronto/Lobby. Messages are synced both ways with sameroom.io.
